CollisionOperator/CollisionProcessors/neutronCEimp_class.f90

```diff
@@ -485,10 +517,20 @@ subroutine split(self, p, thisCycle)
     class(neutronCEimp), intent(inout)    :: self
     class(particle), intent(inout)        :: p
     class(particleDungeon), intent(inout) :: thisCycle
-    integer(shortInt)                     :: mult,i
+    integer(shortInt)                     :: mult, i, n
+    real(defReal)                         :: prob
+
+    n = floor(p % w/self % maxWgt)
+    prob = p % w/self % maxWgt - n
+
+    if (p % pRNG % get() < prob) then
+      mult = n + 1
+    else
+      mult = n
+    end if
+
+    if (mult == 1) return
```

Comment:
I have a vague recollection that similar change was in the weight windows PR and we decided to stay with the `ceiling` approach. Or am I forgetting something?

Reply:
No, I think I changed the two branches the same way but I'll go back to normal!